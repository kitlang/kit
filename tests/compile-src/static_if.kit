function main() {
    static if true {
        puts("true");
    } else {
        puts("???");
    }

    static if false {
        puts("???");
    } else {
        puts("not false");
    }

    var x = true;
    static if x {
        puts("x = true");
    } else {
        puts("x = false");
    }

    // x = false;
    // static if x {
    //     puts("x = true");
    // } else {
    //     puts("x = false");
    // }

    static if 1 == 1 {
        puts("1. 1 == 1");
    } else {
        puts("1. 1 != 1");
    }

    static if 1 != 1 {
        puts("2. 1 != 1");
    } else {
        puts("2. 1 == 1");
    }

    static if 1 == 1 {
        puts("3. 1 == 1");
    }

    static if 1 != 1 {
        puts("4. 1 != 1");
    }

    printf("%i\n", static 1 + 1);
}
