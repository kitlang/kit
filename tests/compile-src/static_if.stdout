true
not false
x = true
1. 1 == 1
2. 1 == 1
3. 1 == 1
2
